components/taxonomy/src/adapters/adapter.rs

```diff
@@ -0,0 +1,364 @@
+    /// - `"heater/temperature-c"` - **GOOD** we know the feature, the device and the unit.
+    /// - `"camera/image"` - **OK** slightly ambiguous, as the camera could decide of a format,
+    ///      a resolution, etc.
+    /// - `"camera/image-png"` - **GOOD** we know the feature, the device and the format. Note that
+    ///      we still don't know the resolution.
+    /// - `"mydevice/warpspeed"` - **BAD** that doesn't look standard, prefer `"x-mydevice/x-warpspeed"`.
+    ///
+    /// # Good descriptions
+    ///
+    /// - "heater/temperature-c" - all the data is provided to avoid accidents.
+    /// - "light/is-on"
+    pub implements: Vec<&'a str>,
+
+    /// Tags describing the feature. Tags are freeform and can also be setup by the user or
+    /// third-party applications through the REST API.
+    ///
+    /// Example tags: "location: bedroom", "use: burglar alarm", "turn me off every night", ...
+    pub tags: Vec<&'a str>,
+
+    /// A unique id for this instance of the feature. This id must be unique to the system.
+    pub id: Id<FeatureId>,
+
+    /// The service to which this feature belongs. This must be an already-registered service.
+    pub service: Id<ServiceId>,
+
+    /// If `Some(sig)`, the feature supports sending values to the device. For instance, a
+    /// heater supports sending desired temperatures, a light supports sending "on" or "off".
+    /// On the other hand, most sensors do not support sending.
+    ///
+    /// Most `send` features do not return a value. However, some `send` features may use values,
+    /// for instance to provide feedback about the send (e.g. time it took for sending, state
+    /// of the device, warnings, etc.).
+    pub send: Option<Signature>,
+
+    /// If `Some(sig)`, the feature supports fetching values to the device. For instance, a
+    /// temperature sensors supports fetching the temperature, but a text-to-speech device
+    /// typically does not support fetching.
+    ///
+    /// Most `fetch` features do not accept an argument. However, some `fetch` features may
+    /// accept an argument, for instance to provide details about how the value should be sent
+    /// (e.g. )
+    pub fetch: Option<Signature>,
+
+    /// If `Some(sig)`, the feature supports deleting. For instance, a camera supports deleting
+    /// values, many devices support deleting custom preferences, but an oven typically does not
+    /// support deleting a temperature.
+    pub delete: Option<Signature>,
+
+    /// If `Some(sig)`, the feature supports watching changes to the device. For instance, a
+    /// motion detector supports watching for motion being detected, or a thermometer for watching
+    /// if temperature rises above a certain value.
+    ///
+    /// The signature's `accepts` represents the condition on values.
+    pub watch: Option<Signature>,
+}
+
+
+impl<'a> Feature<'a> {
+    /// Define an empty feature
+    pub fn empty(id: &Id<FeatureId>, service: &Id<ServiceId>) -> Self {
```

Comment:
Well, I can rename it `default`, but I still need to provide a unique `Id` for `id` and `service`.